src/util/tools.cc

```diff
@@ -326,14 +326,29 @@ static const char* HEX_CHARS2 = "0123456789ABCDEF";
 std::string url_escape(const std::string& str)
 {
```

Comment:
Call me lazy but having this stuff in our code seems like a waste of effort. Is it not in one of our libs? curl maybe?

Reply:
I'm sure it is. But this function is also used to store auxdata in database. That's how I came across it: One of my test files had umlauts in the composer TCOM.

The only way to solve this would be to migrate auxdata to its own table like metadata.

Reply:
I agree, but curl is optional.

About auxdata: does it need to be stored in the db in first place? It's not searchable and the only use case I remember is for import script.

On a side note I'm currently exploring the idea to store resources in a separate table like metadata. Could simplify code significantly especially handlers' part. Also had rid of current 4 different ways to have fanart...